### PR TITLE
testing a script in backend

### PR DESCRIPTION
**Pull request by Rekik27**
Wrote a test script that makes sure api/getData endpoint of backend Flask is running on http://localhost:5000.
Validates the response from the API.

### CODE REVIEW

**Review by jshilts657 — COMMENTED** [review]
@Rekik27 please mark this  pull request as a draft we are going to have to make some changes here

**Comment by jshilts657** [comment]
> @Rekik27 please mark this pull request as a draft we are going to have to make some changes here

image

File: code/backend/testscript.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#from flask_cors import CORS
+#from flask import Flask, request, jsonify# Adjust according to your structure
+
+#app = Flask(__name__)
+#CORS(app)
+
+#@app.route('/api/getData', methods = ['POST'])
+
```
Comment (jshilts657) [inline code/backend/testscript.py]: Try something like this 

@app.route('/login', methods=['POST', 'GET'])
def login():
    error = None
    if request.method == 'POST':
        if valid_login(request.form['username'],
                       request.form['password']):
            return log_the_user_in(request.form['username'])
        else:
            error = 'Invalid username/password'
    # the code below is executed if the request method
    # was GET or the credentials were invalid
    return render_template('login.html', error=error)

**Review by jshilts657 — CHANGES_REQUESTED** [review]
The test script needs to call the getData Function in the python backend

The script should setup a python flask application that does not run on the same port as our backend server. 
The script when started should make a rest api call to the python backends getData function
When that data is returned to the test script it should validate the response.

File: code/backend/testscript.py (hunk 1 above)
Comment (Rekik27) [reply code/backend/testscript.py]: fixed the test script

**Review by jshilts657 — COMMENTED** [review]
ok this is looking much closer, well done. 

How did you test and verify this change?

**Comment by Rekik27** [comment]
> ok this is looking much closer, well done.
> 
> How did you test and verify this change?

I run the test code locally against the backend app and verified.
image

**Review by acflinch — APPROVED** [review]
looks good

**Pull request closed without merging**